File: SafetyCriticalSystem/Code/modules/Module7/Module7.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int readConstant(){
    fstream newfile;
    newfile.open("./modules/Module7/config.txt");
    int res = 0;
    if(newfile.is_open()){
        string tp;
        getline(newfile, tp);
        res = stoi(tp);
        newfile.close();
    }
    return res;
}

void func(int n, int v){
    if(n < v){
        func(n-1, v);
    }
}

int main(int argc, char *argv[]){
    int n = readConstant();
    int v = stoi(argv[1]);
    func(n,v);
}

File: SafetyCriticalSystem/Code/modules/Module7/Module7.java
import java.io.File;
import java.util.Scanner;

public class Module7{
    public static void func(int n, int v){
        if(n < v){
            func(n-1, v);
        }
    }

    public static int readFile(){
        int returnVal = 0;
        try {
            File myObj = new File("./modules/Module7/config.txt");
            Scanner myReader = new Scanner(myObj);
            while (myReader.hasNextLine()) {
                String data = myReader.nextLine();
                returnVal = Integer.parseInt(data);
                break;
            }
            myReader.close();
        } catch (Exception e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }
        return returnVal;
    }

    public static void main(String[] args) {
        int n = readFile();
        int v = Integer.parseInt(args[0]);
        func(n,v);
    }
}